### PR TITLE
feat: support TF 2.2

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
We now support TF 2.2 and have made it the default TF2 version. 

[DET-3088](https://determinedai.atlassian.net/browse/DET-3088)

## Test Plan
Tested manually and ran a full suite of tests: https://app.circleci.com/pipelines/github/determined-ai/determined/1735/workflows/041781c3-3e21-4758-82ef-4b52861e928b

### CODE REVIEW

File: docs/how-to/custom-env.txt [hunk 1]
```diff
@@ -105,8 +105,8 @@ This can be configured in your experiment configuration like below:
 
    environment:
      image:
-       gpu: "determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.1-gpu-0.4.0"
-       cpu: "determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.1-cpu-0.4.0"
+       gpu: "determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-0.4.0"
```
Comment (brain-good) [inline docs/how-to/custom-env.txt:108]: blocking: gotta change these two lines as well!

https://github.com/determined-ai/determined/blob/cb8f4a82b1ac44850aa63fdc7498beacd741321e/docs/how-to/custom-env.txt#L93
https://github.com/determined-ai/determined/blob/cb8f4a82b1ac44850aa63fdc7498beacd741321e/docs/how-to/custom-env.txt#L99

File: harness/determined/keras/_tf_keras_trial.py [hunk 2]
```diff
@@ -112,6 +124,10 @@ def on_train_batch_end(self, _: int, logs: Any = None) -> None:
 
         self.tf_keras_trial_controller.run()
 
+        if self.model.stop_training and version.parse(tf.__version__) >= version.parse("2.0.0"):
+            # Starting with TF 2.2, `model.stop_training` is only checked at the end of epochs.
```
Comment (brain-good) [inline harness/determined/keras/_tf_keras_trial.py:128]: question: it's a little odd to me that the comment is talking about 2.2 but the check is for 2.0. Was this a mistake?

File: docs/how-to/custom-env.txt (hunk 1 above)
Comment (aaron276h) [reply docs/how-to/custom-env.txt:108]: Good call updated.

File: harness/determined/keras/_tf_keras_trial.py (hunk 2 above)
Comment (aaron276h) [reply harness/determined/keras/_tf_keras_trial.py:128]: Oh yeah this is wrong, thanks for catching it.

**Pull request merged**